### ISSUE TITLE
Azure ML inference cluster breaks after update to 3.29

### ISSUE THREAD

**Issue by sdebruyn** [hashicorp/terraform-provider-azurerm#19075]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.7

### AzureRM Provider Version

3.29.1

### Affected Resource(s)/Data Source(s)

azurerm_machine_learning_inference_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_machine_learning_inference_cluster" "redacted" {
  name                          = "redacted"
  location                      = local.location
  tags                          = local.common_tags
  cluster_purpose               = var.environment == "prod" ? "FastProd" : "DevTest"
  kubernetes_cluster_id         = azurerm_kubernetes_cluster.redacted.id
  description                   = "Default Cluster for Redacted"
  machine_learning_workspace_id = azurerm_machine_learning_workspace.data_platform.id
}
```
```


### Debug Output/Panic Output

```shell
│ Error: parsing "/subscriptions/***/resourcegroups/rg-redacted/providers/Microsoft.ContainerService/managedClusters/aks-redacted": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
```

### Expected Behaviour

This worked in version 3.27.0 and before (the config hasn't changed in months), it broke after updating to 3.29

### Actual Behaviour

Error when trying to run  `terraform plan`

### Steps to Reproduce

`terraform plan`

**Comment by aristosvo** [comment]
Hi @sdebruyn! Is this using an imported AKS cluster? I'm very interested in how it ended up like this to make sure it will be fixed in the right way _when it's fixed_.

When updating the `azurerm_kubernetes_cluster` resource to a new API SDK for `3.29`, we moved to a more strict validation pattern for the `azurerm_machine_learning_inference_cluster` as it is/was using the same SDK under the hood.

The quickest but dirtiest would probably be in case of an imported cluster to reimport it with a solid ID or change it in state..

**Comment by sdebruyn** [comment]
The cluster was created using Terraform v2.something

**Comment by djsly** [comment]
@aristosvo  we are seeing it as well for AKS node pool resources : 
```
 Error: parsing "/subscriptions/<subId>/resourcegroups/<rg-name>/providers/Microsoft.ContainerService/managedClusters<cluster-name>/agentPools/win1": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
 ```

**Comment by sdebruyn** [comment]
I did `terraform state rm address-of-the-cluster` and `terraform import address-of-the-cluster ID-of-the-cluster-as-found-in-Azure-portal` and then ran `terraform plan` again, but I got the same error.

**Comment by sdebruyn** [comment]
I tried the same for the inference cluster, but there the import failed with the same error as the one we get in `terraform plan`

**Comment by djsly** [comment]
For us. We simply edited the state file manually to put back the right CamelCase. It worked after that On Nov 3, 2022, at 5:42 AM, Sam Debruyn ***@***.***> wrote:﻿
I tried the same for the inference cluster, but there the import failed with the same error as the one we get in terraform plan

**Comment by sdebruyn** [comment]
I just downgraded to 3.27.0 for now and that version works without issues

**Comment by stephybun** [comment]
@sdebruyn thanks for raising this issue. 

Are you still getting this error using the latest version of the provider? I think this should have been fixed by #20237 in v3.43.0 onwards where we parse the Kubernetes Cluster ID insensitively.

**Comment by stephybun** [comment]
Closing since we haven't heard back. If this is still an issue please feel free to reach out and we can reopen this. Thanks!
